modules/pam_namespace/pam_namespace.c

```diff
@@ -39,6 +39,71 @@
 #include "pam_namespace.h"
 #include "argv_parse.h"
 
+/* --- evaluting all files in VENDORDIR/security/namespace.d and /etc/security/namespace.d --- */
+static int
+compare_filename(const void *a, const void *b)
+{
+	char *a_dup = strdup(*(const char * const *)a);
+	char *b_dup = strdup(*(const char * const *)b);
+	int ret = strcmp(basename(a_dup),
+			 basename(b_dup));
+	free(a_dup);
+	free(b_dup);
+	return ret;
+}
```

Comment:
As no memory allocation is needed here, please find a way to avoid it, consider using strrchr.